cluster/src/main/java/com/hartwig/pipeline/calling/somatic/SageV2Caller.java

```diff
@@ -0,0 +1,124 @@
+
+    private OutputFile outputFile;
+    private OutputFile sageOutputFile;
+
+    public SageV2Caller(final AlignmentPair alignmentPair) {
+        super(alignmentPair);
+    }
+
+    @Override
+    public String namespace() {
+        return NAMESPACE;
+    }
+
+    @Override
+    public List<BashCommand> commands(final SomaticRunMetadata metadata) {
+
+        List<BashCommand> commands = Lists.newArrayList();
+        commands.add(new UnzipToDirectoryCommand(VmDirectories.RESOURCES, Resource.SNPEFF_DB));
+
+        String tumorBamPath = getTumorBamDownload().getLocalTargetPath();
+        String referenceBamPath = getReferenceBamDownload().getLocalTargetPath();
+        String referenceGenomePath = Resource.REFERENCE_GENOME_FASTA;
+        String tumorSampleName = metadata.tumor().sampleName();
+        String referenceSampleName = metadata.reference().sampleName();
+
+        SageV2Application sageV2Application = new SageV2Application(Resource.of(SAGE, "KnownHotspots.hg19.vcf.gz"),
+                Resource.of(SAGE, "ActionableCodingPanel.hg19.bed.gz"),
+                Resource.of(BEDS, "NA12878_GIAB_highconf_IllFB-IllGATKHC-CG-Ion-Solid_ALLCHROM_v3.2.2_highconf.bed"),
+                referenceGenomePath,
+                tumorBamPath,
+                referenceBamPath,
+                tumorSampleName,
+                referenceSampleName);
+        sageOutputFile = sageV2Application.apply(SubStageInputOutput.empty(tumorSampleName)).outputFile();
+
+        SubStageInputOutput sageOutput = sageV2Application
+                .andThen(new SageV2PassFilter(tumorSampleName))
+                .andThen(new MappabilityAnnotation(Resource.of(MAPPABILITY, "out_150_hg19.mappability.bed.gz"), Resource.of(MAPPABILITY, "mappability.hdr")))
+                .andThen(new PonAnnotation("sage.pon", Resource.of(SAGE, "SageGermlinePon.hg19.vcf.gz"), "PON_COUNT"))
+                .andThen(new SageV2PonFilter())
+                .andThen(new SnpEff(Resource.SNPEFF_CONFIG))
+                .andThen(new SageV2PostProcess("hg19"))
+                .andThen(FinalSubStage.of(new CosmicAnnotation(Resource.COSMIC_VCF_GZ, "ID,INFO")))
+                .apply(SubStageInputOutput.empty(tumorSampleName));
+
+        commands.addAll(sageOutput.bash());
+        outputFile = sageOutput.outputFile();
+        return commands;
+    }
+
+    @Override
+    public VirtualMachineJobDefinition vmDefinition(final BashStartupScript bash, final ResultsDirectory resultsDirectory) {
+        return VirtualMachineJobDefinition.sageCalling(bash, resultsDirectory);
+    }
+
+    @Override
+    public SageV2CallerOutput output(final SomaticRunMetadata metadata, final PipelineStatus jobStatus, final RuntimeBucket bucket,
+            final ResultsDirectory resultsDirectory) {
+        return SageV2CallerOutput.builder()
+                .status(jobStatus)
+                .maybeFinalSomaticVcf(GoogleStorageLocation.of(bucket.name(), resultsDirectory.path(outputFile.fileName())))
```

Comment:
For now this isn't strictly required as I suppose Purple (or anything else downstream) won't be using this VCF .